### DISCUSSION TITLE
Multiply Identity / RETURNING on Update/Insert

### DISCUSSION THREAD

**Discussion by RegBox22** [linq2db/linq2db#3326] [Q&A] [answered]
Good day!
 Please tell me how you can combine insert / update and selection into one request for reading COMPUTED columns via RETUTNING for FirebirdSql based on Identity of field descriptions.

```
[Table (“TEST”)]
public class TEST 
{
  [PrimaryKey, Identity]
  public int ID {get; set; } // Increment With Triger
  
  [Column]
  public int LEVEL {get; set; }
  
  [Identity)] // ver 1
  // [Column (SkipOnInsert = true, SkipOnUpdate = true)] // ver 2
  public int LEVEL_PLUS_ONE {get; set; } // Computed (LEVEL + 1)
}

```
The task is to get Linq analogs of a call in one request
**Insert**
```
{
       var newId = (long) await db.InsertWithIdentityAsync (testEntity);
       var result = await db.TEST.FirstOrDefaultAsync (x => x.ID == newId);
}
```
**Update**
```
{
await db.UpdateAsync (test);
       var result = await db.TEST.FirstOrDefaultAsync (x => x.ID == test.ID);
}
```
**Into something similar**
```
{
var result = from t in db.TEST
                    where t.ID == (db.TEST.InsertWithIdentityAsync (testEntity))
                    select t;
}
```
```
// insert into TEST (LEVEL) values ​​(@ LEVEL_1) RETURNING ID, LEVEL_PLUS_ONE
// update TEST set LEVEL = @ LEVEL_1 where ID = @ ID RETURNING LEVEL_PLUS_ONE
```

Earlier, in its CrudHelper (ADO.NET), I used RETURNING for all table fields, but in LinqToDb, apparently one Identity field is used, and the last one specified in the table if the LEVEL_PLUS_ONE field is wrapped in [Identity] then InsertWithIdentityAsync will return RETURNING LEVEL_PLONE_ONE, not RETURNING ID

**Comment by MaceWindu** [comment, marked as answer]
I don't think you can do it rigth now. Currently `*WithOutput` APIs we have generate proper SQL only for SQL Server database. I will check if it is possible to extend them easily to work with other databases with `returning`/`output` clause...

**Reply by MaceWindu** [reply]
Pending PR: #3328
